### PR TITLE
Fix issue wso2/product-iot 1184

### PR DESCRIPTION
**Pull request by dilee**
## Purpose
Fixing issue wso2/product-iot 1184 - Redirection not handled upon session expiration.

### CODE REVIEW

**Review by madhawap — APPROVED** [review]

File: components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/modules/business-controllers/device.js [hunk 1]
```diff
@@ -300,7 +306,9 @@ deviceModule = function () {
 			);
 		} else {
 			log.error("User object was not found in the session");
-			throw constants["ERRORS"]["USER_NOT_FOUND"];
+            userModule.logout(function () {
```
Comment (prabushi) [inline components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/modules/business-controllers/device.js:309]: Fix formatting issue.

File: components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/modules/business-controllers/device.js [hunk 2]
```diff
@@ -369,7 +377,9 @@ deviceModule = function () {
 		var carbonUser = session.get(constants["USER_SESSION_KEY"]);
 		if (!carbonUser) {
 			log.error("User object was not found in the session");
-			throw constants["ERRORS"]["USER_NOT_FOUND"];
+            userModule.logout(function () {
```
Comment (prabushi) [inline components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/modules/business-controllers/device.js:380]: Fix formatting issue.

File: components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/modules/business-controllers/device.js [hunk 3]
```diff
@@ -261,7 +263,9 @@ deviceModule = function () {
                 url, function (responsePayload) {
                     if(!responsePayload["responseText"]){
```
Comment (prabushi) [inline components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/modules/business-controllers/device.js:264]: Fix the indentation issues.

File: components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/modules/business-controllers/device.js (hunk 1 above)
Comment (dilee) [reply components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/modules/business-controllers/device.js:309]: Fixed in https://github.com/wso2/carbon-device-mgt/pull/1275

File: components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/modules/business-controllers/device.js (hunk 2 above)
Comment (dilee) [reply components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/modules/business-controllers/device.js:380]: Fixed in https://github.com/wso2/carbon-device-mgt/pull/1275

File: components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/modules/business-controllers/device.js (hunk 3 above)
Comment (dilee) [reply components/device-mgt/org.wso2.carbon.device.mgt.ui/src/main/resources/jaggeryapps/devicemgt/app/modules/business-controllers/device.js:264]: Fixed in https://github.com/wso2/carbon-device-mgt/pull/1275

**Pull request merged**